checker/src/abstract_value.rs

```diff
@@ -2537,16 +2559,126 @@ impl AbstractValueTrait for Rc<AbstractValue> {
     /// Constructs an element of the tag domain for simple expressions.
     #[logfn_inputs(TRACE)]
     fn get_tags(&self) -> TagDomain {
-        // todo: take into consideration the control masks for different tags.
-        // todo: for now simply return an empty set (unsound).
-        TagDomain::for_empty_set()
+        let tag_id_opt = self.expression.get_tag_identifier();
+
+        // First deal with expressions that do not propagate tags or have special propagation behavior.
+        match &self.expression {
+            Expression::Top
+            | Expression::Bottom
+            | Expression::CompileTimeConstant { .. }
+            | Expression::HeapBlock { .. }
+            | Expression::HeapBlockLayout { .. }
+            | Expression::Reference { .. }
+            | Expression::UnknownModelField { .. }
+            | Expression::Variable { .. } => {
+                return TagDomain::for_empty_set();
+            }
+
+            Expression::ConditionalExpression {
+                condition,
+                consequent,
+                alternate,
+            } => {
+                // tags(condition) | (tags(consequent) join tags(alternate))
+                return condition.get_cached_tags().union(
```

Comment:
I'm a bit surprised to see get_cached_tags here. I fully expected to see get_tags. Can you explain why?

Reply:
There are two reasons:
1. The subexpressions might already have their tag field computed, so we don't need to recompute from scratch.
2. For now there is no tagged expressions, so it seems impossible to reconstruct tags from the expression itself.

Reply:
1. I confused get_cached_tags with get_tags, so all is good.